src/safe.js

```diff
@@ -127,448 +27,533 @@ export default function createSafeModule(
+        data: await safeSdk
+          .createTransaction({
+            safeTransactionData: {
+              to: safeAddress,
+              value: 0,
+              data: safeInstance.methods
+                .setFallbackHandler(fallbackHandlerAddress)
+                .encodeABI(),
+            },
+          })
+          .then((safeTx) =>
+            _prepareSafeTransaction({
+              safeTx,
+              safeSdk,
+              signerAddress: account.address,
+            }),
+          ),
       });
 
-      const safeVersion = await getVersion(web3, options.safeAddress);
-      let txHashChangeMasterCopy;
-      let txHashFallbackHandler;
-
-      if (safeVersion != SAFE_LAST_VERSION) {
-        // References:
-        // https://github.com/safe-global/web-core/blob/main/src/services/tx/safeUpdateParams.ts
-        // https://github.com/safe-global/safe-react/blob/main/src/logic/safe/utils/upgradeSafe.ts
-
-        // Get the Safe contract with version v1.1.1+Circles
-        const safeInstance = getSafeCRCVersionContract(
-          web3,
-          options.safeAddress,
-        );
-
-        // First we change the Master Copy to v1.3.0
-        // @ts-expect-error this was removed in 1.3.0 but we need to support it for older safe versions
-        const updateSafeTxData = safeInstance.methods
-          .changeMasterCopy(safeMaster.options.address)
-          .encodeABI();
-        txHashChangeMasterCopy = await utils.executeTokenSafeTx(account, {
-          safeAddress: options.safeAddress,
-          to: options.safeAddress,
-          txData: updateSafeTxData,
-          isCRCVersion: true,
-        });
-        if (!txHashChangeMasterCopy) {
-          throw new CoreError(
-            `Safe with version ${safeVersion} failed to change the Master Copy`,
-          );
-        }
-
-        // Wait to check that the version is updated
-        await loop(
-          () => {
-            return getVersion(web3, options.safeAddress);
-          },
-          (version) => {
-            return version == SAFE_LAST_VERSION;
-          },
-        );
-
-        // Then we setup the fallbackHandler
-        const fallbackHandlerTxData = safeInstance.methods
-          .setFallbackHandler(fallbackHandlerAddress)
-          .encodeABI();
-        txHashFallbackHandler = await utils.executeTokenSafeTx(account, {
-          safeAddress: options.safeAddress,
-          to: options.safeAddress,
-          txData: fallbackHandlerTxData,
-        });
-      }
-      return { txHashChangeMasterCopy, txHashFallbackHandler };
-    },
+      // Wait to check that the version is updated
+      safeVersion = await loop(
+        () => _getVersion(safeAddress),
+        (version) => version === SAFE_LAST_VERSION,
+        { label: 'Waiting for CRC Safe to upgrade version' },
+      );
+    }
+
+    return safeVersion;
+  };
+
+  // TODO: this method is missing to be implemented because it will be moved and replaced into organization.js.
```

Comment:
should we remove it regardeless?

Reply:
It is still there to remind that change in the futuro and do not forget.